### PR TITLE
Implement backend part for game Rock Paper Scissors

### PR DESCRIPTION
**Pull request by yuferevaleksey**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,73 +1,207 @@
...
+```javascript
+interface GotToNextRoundMessage {
+  gameId: string;
+  socketId: string;
+}
+```
 
-## License
+#### Responce:
+See: *OutgoingEvents.GAME_RESPONSE*
```
Comment (Kimmi-wunderdog) [inline README.md:215]: As this is an educational project, I would recommend you to add some links in this file, for all the technologies used.

File: src/rock-paper-scissors/services/game-process/shapes/rock.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { ShapeAbstract } from './shape-abstract';
+import { PlayerChoice, Shapes } from '../../../types/game';
+
+export class Rock extends ShapeAbstract {
+  achillesHeel: Shapes = Shapes.PAPER;
```
Comment (Kimmi-wunderdog) [inline src/rock-paper-scissors/services/game-process/shapes/rock.ts]: Variable name is a bit missleading to me

File: src/rock-paper-scissors/services/game.service.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
+import { Injectable } from '@nestjs/common';
+import { GameModel } from './game.model';
+import { Game } from '../schemas/game.schema';
+import { GameItem, GameRound, PlayerChoice } from '../types/game';
+import { AddChoiceToRoundDto } from '../dto/add-choice-to-round.dto';
+import { GameProcess } from './game-process/game-process.service';
+
+@Injectable()
+export class GameService {
+  readonly NO_ONE_WIN = 'noOneWin';
```
Comment (Kimmi-wunderdog) [inline src/rock-paper-scissors/services/game.service.ts]: Let's be ready for current sensibilities, better say TIE :D

File: src/rock-paper-scissors/services/game.service.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+   * @param socketId
+   */
+  public async joinGame(
+    gameId: string,
+    nickName: string,
+    socketId: string,
+  ): Promise<Game> {
+    const gameData = await this.gameModel.findGame(gameId);
+    if (!gameData) {
+      // @todo: throw error here;
```
Comment (Kimmi-wunderdog) [inline src/rock-paper-scissors/services/game.service.ts]: It seems you missed this not covered TODO

File: src/rock-paper-scissors/services/game.service.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+   * @param gameId
+   * @param playerChoice
+   */
+  public async makeChoice(
+    gameId: string,
+    playerChoice: PlayerChoice,
+  ): Promise<Game> {
+    const gameData = await this.gameModel.findGame(gameId);
+    if (!gameData) {
+      // @todo: throw error here;
```
Comment (Kimmi-wunderdog) [inline src/rock-paper-scissors/services/game.service.ts]: Same as above

File: src/rock-paper-scissors/services/game.service.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+    return gameData;
+  }
+
+  /**
+   * Get list for all games.
+   */
+  public async getGamesList(): Promise<GameItem[]> {
+    const gamesList = await this.gameModel.findAll();
+    return gamesList
```
Comment (Kimmi-wunderdog) [inline src/rock-paper-scissors/services/game.service.ts]: Can we do this filter at query level? I have the concern that it make be heavy in memory if the game become famous

File: src/rock-paper-scissors/services/game-process/shapes/rock.ts (hunk 2 above)
Comment (yuferevaleksey) [reply src/rock-paper-scissors/services/game-process/shapes/rock.ts]: Heh,  https://en.wikipedia.org/wiki/Achilles%27_heel
That is a metaphor to shape weaknesses, each shape has it, Rock has Paper as a weakness, and so on

Comment (yuferevaleksey) [reply src/rock-paper-scissors/services/game-process/shapes/rock.ts]: I will rename in to **weakness**

File: src/rock-paper-scissors/services/game.service.ts (hunk 3 above)
Comment (yuferevaleksey) [reply src/rock-paper-scissors/services/game.service.ts]: Sure.

File: src/rock-paper-scissors/services/game.service.ts (hunk 6 above)
Comment (yuferevaleksey) [reply src/rock-paper-scissors/services/game.service.ts]: Heh, become famous, fanny. 
But yep, you are right, that could be easily moved to query level.

File: README.md (hunk 1 above)
Comment (yuferevaleksey) [reply README.md:215]: Sure.

File: src/rock-paper-scissors/services/game.service.ts (hunk 4 above)
Comment (yuferevaleksey) [reply src/rock-paper-scissors/services/game.service.ts]: Fixed

File: src/rock-paper-scissors/services/game.service.ts (hunk 5 above)
Comment (yuferevaleksey) [reply src/rock-paper-scissors/services/game.service.ts]: Fixed

**Pull request merged**